# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContext.h
# Commit: 9ff26bcc580c
# Full Hash: 9ff26bcc580cf8e25559f9911a9fc860baf8d4fc
# Author: Imanol Fernandez <mortimergoro@gmail.com>
# Date: 2020-06-09 09:21:34
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Create a separate WebGLContext::PresentInto function for WebXR. r=jgilbert
#   
#   - Do not apply mIsCanvasDirty bailout when presenting from a opaque FBO back buffer.
#   - Create a separate WebGLContext::PresentInto with more specific XR validations and logic.
#   - Use the Opaque FBO backbuffer as source when calling BlitBackbufferToCurDriverFB
# ==============================================================================

diff -r 16d84439756f -r 9ff26bcc580c dom/canvas/WebGLContext.h
--- a/dom/canvas/WebGLContext.h	Mon Jun 08 20:36:14 2020 +0000
+++ b/dom/canvas/WebGLContext.h	Mon Jun 08 20:35:38 2020 +0000
@@ -101,7 +101,7 @@
 namespace layers {
 class CompositableHost;
 class SurfaceDescriptor;
-}
+}  // namespace layers
 
 namespace webgl {
 class AvailabilityRunnable;
@@ -483,6 +483,7 @@
   // Present to compositor
  private:
   bool PresentInto(gl::SwapChain& swapChain);
+  bool PresentIntoXR(gl::SwapChain& swapChain, const gl::MozFramebuffer& fb);
 
  public:
   void Present(WebGLFramebuffer*, layers::TextureType);
